test/common/storagelib.py

```diff
@@ -19,18 +19,33 @@
 import os.path
 import re
 import textwrap
+from typing import Any, Protocol
 
-from testlib import Error, MachineCase, wait
+from machine import testvm
+from testlib import Browser, Error, MachineCase, wait
 
 
-def from_udisks_ascii(codepoints):
+def from_udisks_ascii(codepoints: list[int]) -> str:
     return ''.join(map(chr, codepoints[:-1]))
 
 
-class StorageHelpers:
+class StorageMachineProtocol(Protocol):
+    """Classes that StorageHelpers is mixed into need to implement this"""
+
+    machine: testvm.Machine
+    browser: Browser
+
+    def allow_browser_errors(self, *patterns: str) -> None: ...
+
+
+class StorageHelpers(StorageMachineProtocol):
```

Comment:
Something seems a bit off here.  The purpose of a Protocol is to enable structural typing in cases where you specifically *don't* want to subclass the protocol type.  Here the first thing you do after creating it is to immediately subclass it...

Reply:
Hmmmm, I don't see another way to type mixins? Nor is this a real protocol anyway.

Reply:
I don't think you need to do anything special with the typing of mixins.

Protocols are really for the case where you have some method that takes an argument and that argument can be anything at all, so long as some condition holds on it, like "has a `.close()` function I can call".  You could create a `Closable` interface and declare a `def close(self):` on it, and force everyone to formally implement/derive that, but this is annoying and unpythonic: it should be possible to pass any instance of a class that has a `close()` method without having this silly subclassing requirement.  That's what a protocol is.  You define an interface and then anything that structurally satisfies the interface can be passed an an argument of that type, even if it doesn't formally derive from it.

Reply:
Here's the docs:

https://mypy.readthedocs.io/en/stable/protocols.html

There's no mention about mixins in there anywhere...